client/rpc-spec/src/chain_spec/chain_spec.rs

```diff
@@ -0,0 +1,68 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! API implementation for the specification of a chain.
+
+use crate::chain_spec::api::ChainSpecApiServer;
+use jsonrpsee::core::RpcResult;
+use sc_chain_spec::Properties;
+
+/// An API for chain spec RPC calls.
+pub struct ChainSpec {
+	/// The name of the chain.
+	name: String,
+	/// The hexadecimal encoded hash of the genesis block.
+	///
+	/// # Note
+	///
+	/// This can be empty, or prefixed with "0x".
```

Comment:
Why can this be empty? There's never a situation where we don't know the genesis hash.

Reply:
It makes sense, indeed I was thinking of this to receive untrusted/malformed input. Given we control that, will make sure to always format it as `0x[hash]`. Thanks for spotting this out!

Reply:
Have pushed a fix for it! Thanks again!